### PR TITLE
NIFI-10750 Move C2NiFiClientService to minifi-framework

### PR DESCRIPTION
**Pull request by ferencerdei**
# Summary
Move the C2NiFiClientService to the same module as the BootstrapListener so we can achieve communication between C2 and minifi Bootstrap.
[NIFI-10750](https://issues.apache.org/jira/browse/NIFI-10750)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [x] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: minifi/minifi-assembly/pom.xml [hunk 1]
```diff
@@ -121,10 +121,6 @@ limitations under the License.
             <groupId>org.apache.nifi.minifi</groupId>
             <artifactId>minifi-framework-api</artifactId>
         </dependency>
-        <dependency>
```
Comment (bejancsaba) [inline minifi/minifi-assembly/pom.xml:124]: Thanks for the cleanup

File: minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-runtime/src/main/java/org/apache/nifi/minifi/bootstrap/BootstrapRequestReader.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    public BootstrapRequestReader(String secretKey) {
+        this.secretKey = secretKey;
+    }
+
+    // we don't want to close the stream, as the caller will do that
+    public BootstrapRequest readRequest(InputStream in) throws IOException {
+        // We want to ensure that we don't try to read data from an InputStream directly
+        // by a BufferedReader because any user on the system could open a socket and send
+        // a multi-gigabyte file without any new lines in order to crash the NiFi instance
```
Comment (bejancsaba) [inline minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-runtime/src/main/java/org/apache/nifi/minifi/bootstrap/BootstrapRequestReader.java:38]: NiFI -> MiNiFi :)

**Review by bejancsaba — APPROVED** [review]
+1, Thank you for the refactor now this logic seems to be much better aligned with the location. It was quite foreign inside nifi core.

**Review by kevdoran — APPROVED** [review]
This looks like a nice improvement, thanks @ferencerdei!

**Pull request closed without merging**